plugins/core/IO/qLASIO/src/LasTiler.cpp

```diff
@@ -0,0 +1,208 @@
+//##########################################################################
+//#                                                                        #
+//#                CLOUDCOMPARE PLUGIN: LAS-IO Plugin                      #
+//#                                                                        #
+//#  This program is free software; you can redistribute it and/or modify  #
+//#  it under the terms of the GNU General Public License as published by  #
+//#  the Free Software Foundation; version 2 of the License.               #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#                   COPYRIGHT: Thomas Montaigu                           #
+//#                                                                        #
+//##########################################################################
+
+#include "LasTiler.h"
+
+#include <ccProgressDialog.h>
+
+#include <QFileInfo>
+#include <QElapsedTimer>
+
+CC_FILE_ERROR TileLasReader(laszip_POINTER laszipReader, const QString &originName, const LasTilingOptions &options) {
+    laszip_header *laszipHeader{nullptr};
+    laszip_CHAR *errorMsg{nullptr};
+
+    if (laszip_get_header_pointer(laszipReader, &laszipHeader)) {
+        laszip_get_error(laszipHeader, &errorMsg);
+        ccLog::Warning("[LAS] laszip error: '%s'", errorMsg);
+        laszip_close_reader(laszipReader);
+        laszip_clean(laszipReader);
+        laszip_destroy(laszipReader);
+        return CC_FERR_THIRD_PARTY_LIB_FAILURE;
+    }
+
+    laszip_U64 pointCount;
+    if (laszipHeader->version_minor == 4) {
+        pointCount = laszipHeader->extended_number_of_point_records;
+    } else {
+        pointCount = laszipHeader->number_of_point_records;
+    }
+
+    laszip_point *laszipPoint{nullptr};
+    if (laszip_get_point_pointer(laszipReader, &laszipPoint)) {
+        laszip_get_error(laszipHeader, &errorMsg);
+        ccLog::Warning("[LAS] laszip error: '%s'", errorMsg);
+        laszip_close_reader(laszipReader);
+        laszip_clean(laszipReader);
+        laszip_destroy(laszipReader);
+        return CC_FERR_THIRD_PARTY_LIB_FAILURE;
+    }
+
+    std::vector<laszip_POINTER> writers;
+    writers.resize(options.numTiles0 * options.numTiles1, nullptr);
+
+    CC_FILE_ERROR error = CC_FERR_NO_ERROR;
+
+    const size_t index0 = options.index0();
+    const size_t index1 = options.index1();
+
+    double cloudMins[3] = {
```

Comment:
I guess you can remove the '=' signs here and below?

Reply:
You mean 
```C++
 double cloudMins[3] { ... }
```
?

That is not legal syntax 🤔

Reply:
Ah yes, there's no 'constructor' for C-arrays, my bad. I'm more used to std::array and std::vectors now...